core/src/main/scala/com/microsoft/azure/synapse/ml/causal/ResidualTransformer.scala

```diff
@@ -0,0 +1,83 @@
+package com.microsoft.azure.synapse.ml.causal
+
+import com.microsoft.azure.synapse.ml.codegen.Wrappable
+import com.microsoft.azure.synapse.ml.logging.BasicLogging
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.linalg.{SQLDataTypes, Vector}
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.util.{DefaultParamsReadable, DefaultParamsWritable, Identifiable}
+import org.apache.spark.sql.functions.{col, udf}
+import org.apache.spark.sql.{DataFrame, Dataset}
+import org.apache.spark.sql.types._
+
```

Comment:
import org.apache.spark.ml.functions.vector_to_array

Reply:
Also please cleanup the imports here: we don't need udf from line 9 and Vector from line 6.